src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotifications.java

```diff
@@ -124,43 +124,46 @@ private static void createStatus(@NonNull Run<?, ?> build, @NonNull TaskListener
         }
 
         String name = build.getFullDisplayName(); // use the build number as the display name of the status
-        BitbucketBuildStatus status;
         Result result = build.getResult();
         String buildDescription = build.getDescription();
         String statusDescription;
-        String state;
+        BitbucketBuildStatus.Status state;
         if (Result.SUCCESS.equals(result)) {
             statusDescription = StringUtils.defaultIfBlank(buildDescription, "This commit looks good.");
-            state = SUCCESSFUL_STATE;
+            state = BitbucketBuildStatus.Status.SUCCESSFUL;
         } else if (Result.UNSTABLE.equals(result)) {
             statusDescription = StringUtils.defaultIfBlank(buildDescription, "This commit has test failures.");
-
             BitbucketSCMSource source = (BitbucketSCMSource) s;
             BitbucketSCMSourceContext sourceContext = new BitbucketSCMSourceContext(null, SCMHeadObserver.none())
                     .withTraits(source.getTraits());
             if (sourceContext.sendSuccessNotificationForUnstableBuild()) {
-                state = SUCCESSFUL_STATE;
+                state = BitbucketBuildStatus.Status.SUCCESSFUL;
             } else {
-                state = FAILED_STATE;
+                state = BitbucketBuildStatus.Status.FAILED;
             }
         } else if (Result.FAILURE.equals(result)) {
             statusDescription = StringUtils.defaultIfBlank(buildDescription, "There was a failure building this commit.");
-            state = FAILED_STATE;
+            state = BitbucketBuildStatus.Status.FAILED;
         } else if (Result.NOT_BUILT.equals(result)) {
             // Bitbucket Cloud and Server support different build states.
-            state = (bitbucket instanceof BitbucketCloudApiClient) ? STOPPED_STATE : SUCCESSFUL_STATE;
             statusDescription = StringUtils.defaultIfBlank(buildDescription, "This commit was not built (probably the build was skipped)");
+            state = (bitbucket instanceof BitbucketCloudApiClient) ? BitbucketBuildStatus.Status.STOPPED : null;
```

Comment:
This is the key change.   

```suggestion
            state = (bitbucket instanceof BitbucketCloudApiClient) ? BitbucketBuildStatus.Status.STOPPED : BitbucketBuildStatus.Status.FAILED;
```

The comments below this indicate that `ABORTED` should send a `FAILED` state notification. 
 
It is unclear to me whether `NOT_BUILT` should report failed or no result as you've done here. 

The problem with reporting no result is that it is completely different behavior between cloud and server. 😢 
How does Bitbucket Cloud treat the `STOPPED` status, is kind of success or a kind of failure? 

I would say that if `ABORTED` reports `FAILED` then `NOT_BUILT` should also report `FAILED`.

Reply:
I think not sending a status update, or making it configurable (NOT_BUILT -> FAILED, NOT_BUILT -> SUCCESS, NOT_BUILD -> don't report anything), is the most sensible thing here to do. Mapping it to FAILED would help my use-case (only successful builds should get merged + stop previous builds if new commits came in) but was not tolerable for some other use-cases: https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/110. Not reporting it would help both use-cases I think.

Reply:
Huch. Success for NOT_BUILT jobs was implemented on purpose. I have the same use case as @gregorburger. That was a reason i raised the PR. I'm agree: making it configurable is way to go. But i've no idea, how i can implement it. And it seems, that my PR will break #110. @bitwiseman: Sorry, i can't answer your question about BitBucket Cloud. I have no access to. And yes, my first implementation simply sent FAILED. This way developer were not able to figure out, if there is a problem with PR or the build was simply superseeded.

Reply:
In this PR a UI was added to configure unstable builds https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/379. It seems there are a variety of different use-cases around with regards to build state and more configuration options could  help here.